File: tests/15_cmp_1.py
#!/usr/bin/env python3

from interp import *

def t():
    c = [
        1,
        2,
    ]
    i = [
        cpu_encode_op_lib_i64(0,0),
        cpu_encode_op_lib_i64(1,1),
        cpu_encode_op_cmp_i64(0,1, Fun3Compare.compare_lt.value),
        cpu_encode_op_break(0)
    ]
    cpu_test("cmp_1", c, i)

t()

File: tests/15_cmp_1.go
package main

import . "glyph/tests/interp"

func t() {
    c := []uint64{
        1,
        2,
    }
    i := []uint16{
        CPU_encode_op_lib_i64(0,0),
        CPU_encode_op_lib_i64(1,1),
        CPU_encode_op_cmp_i64(0,1, Compare_lt),
        CPU_encode_op_break(0),
    }
    CPU_test("cmp_1", c, i);
}

func main() {
    t()
}
